examples/plugin/README.md

```diff
@@ -0,0 +1,139 @@
+- [HashiCorp go-plugin](https://github.com/hashicorp/go-plugin) (used by both host and plugin)
+- **PingCLI v0.7.0+** installed and configured
+
+## How Plugins Work
+
+1.  **Discovery**: When `pingcli` starts, it loads a list of registered plugin executables from its configuration profile. This list is managed by the `pingcli plugin` command. Plugins must be on your system `PATH`
+2.  **Handshake**: For each registered plugin, `pingcli` launches the executable as a child process. A secure handshake is performed to verify that the child process is a valid plugin and is compatible with the host.
+3.  **Communication**: Once the handshake is complete, the host and plugin communicate over gRPC. The host can call functions defined in the plugin (like `Run`), and the plugin can send log messages back to the host.
+4.  **Execution**: When a user runs a command provided by a plugin, `pingcli` invokes the corresponding gRPC method in the plugin process, passing along any arguments and flags.
+5. **Compatibility**: The Handshake process includes a ProtocolVersion check. This ensures that the plugin and the pingcli host are compatible, preventing issues if the underlying gRPC interface changes in future versions of pingcli.
+
+## Building a Plugin
+
+1. **Clone or create your plugin source code.**  
+   See [`plugin.go`](plugin.go) for a complete example.
+
+2. **Build the plugin binary:**
+   ```sh
+   go build -o my-plugin
+   ```
+
+3. **Place the binary in a directory on your `PATH`:**
+   ```sh
+   mv my-plugin ~/go/bin/
+   # or any directory in your $PATH
+   ```
+
+## Registering and Managing Plugins
+
+`pingcli` provides the `plugin` command to manage the lifecycle of your plugins.
+
+### Adding a Plugin
+
+To add a new plugin, use the add subcommand. Crucially, the plugin executable must first be placed in a directory that is part of your system's PATH environment variable. pingcli relies on the system's PATH to find the executable to run.
+
+```bash
+pingcli plugin add <executable-name>
+```
+
+### Listing Plugins
+
+To see a list of all currently registered plugins, use the `list` subcommand.
+
+```bash
+pingcli plugin list
+```
+
+### Removing a Plugin
+
+To unregister a plugin from `pingcli`, use the `remove` subcommand.
+
+```bash
+pingcli plugin remove <executable-name>
+```
+
+## Plugin Command Interface
+
+To create a valid plugin, you must implement the `grpc.PingCliCommand` interface. This interface has two methods:
+
+#### `Configuration() (*grpc.PingCliCommandConfiguration, error)`
+
+This method is called by the `pingcli` host to get metadata about your command. This allows `pingcli` to display your command in the help text (`pingcli --help`) and provide detailed help for your specific command (`pingcli your-command --help`) without having to execute its main logic.
+
+The `PingCliCommandConfiguration` struct has the following fields, which correspond directly to properties of a `cobra` command:
```

Comment:
Might be nice to make `cobra` a link to cobra docs

Reply:
Added!